assets/src/edit-story/components/tablist/panel.js

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { useMemo } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+/**
+ * Internal dependencies
+ */
+import { Icons } from '../../../design-system';
+import { PANEL_STATES } from './constants';
+import {
+  Badge,
+  ButtonText,
+  IconContainer,
+  PanelWrapper,
+  SmallHeadline,
+  TabButton,
+  TabPanel,
+} from './styles';
+
+const Panel = ({
+  badgeCount = 0,
+  children,
+  className,
+  isExpanded,
+  onClick,
+  status = PANEL_STATES.NORMAL,
+  title,
+}) => {
+  const panelId = useMemo(uuidv4, []);
+
+  return (
+    <PanelWrapper className={className} isExpanded={isExpanded}>
+      <TabButton
+        aria-controls={panelId}
+        aria-selected={isExpanded}
+        onClick={onClick}
+        role="tab"
+        status={status}
+      >
+        <ButtonText>
+          <IconContainer>
+            <Icons.ChevronDownSmall />
+          </IconContainer>
+          <SmallHeadline id={`${title}-${panelId}`}>{title}</SmallHeadline>
+        </ButtonText>
+        <Badge>
+          <SmallHeadline>{badgeCount}</SmallHeadline>
+        </Badge>
+      </TabButton>
+      <TabPanel aria-labelledby={`${title}-${panelId}`} role="tabpanel">
+        {children}
+      </TabPanel>
+    </PanelWrapper>
+  );
+};
+Panel.propTypes = {
+  badgeCount: PropTypes.number,
+  children: PropTypes.node,
+  className: PropTypes.string,
+  isExpanded: PropTypes.bool,
+  onClick: PropTypes.click,
+  status: PropTypes.oneOf(['normal', 'danger']),
```

Comment:
nit: reuse the `PANEL_STATES` you made - 

```suggestion
  status: PropTypes.oneOf(Object.values(PANEL_STATES)),
```